File: src/food.ts
import * as z from "zod";

export const foodSchema = z.object({
  id: z.number(),
  name: z.string(),
  image: z.string(),
  price: z.number(),
  description: z.string(),
  tags: z.array(z.string()),
});

export type Food = z.infer<typeof foodSchema>;

export type NewFood = Omit<Food, "id">;

export const foodTags = [
  "Breakfast",
  "Lunch",
  "Dinner",
  "Dessert",
  "Drink",
  "Appetizer",
  "Spicy",
  "Vegetarian",
  "Alcoholic",
] as const;

export type FoodTag = (typeof foodTags)[number];

File: backend/Models/Food.cs
using System.ComponentModel.DataAnnotations;

namespace RestaurantApi.Models;

public class Food
{
    public int Id { get; set; }

    [Required]
    [StringLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(200)]
    public string Image { get; set; } = string.Empty;

    [Required]
    [Range(0.01, 9999.99)]
    public decimal Price { get; set; }

    [Required]
    [StringLength(500)]
    public string Description { get; set; } = string.Empty;

    [Required]
    public string[] Tags { get; set; } = Array.Empty<string>();
}
